### PR TITLE
feat: twitter nft project avatar

### PR DESCRIPTION
**Pull request by hanszhoou**
## Description
https://mask.atlassian.net/browse/MF-2873
https://mask.atlassian.net/browse/MF-2211
https://mask.atlassian.net/browse/MF-2893

Closes # (NO_ISSUE)

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

**Comment by guanbinrui** [comment]
image

File: packages/web3-shared/evm/src/helpers/formatter.ts [hunk 1]
```diff
@@ -43,12 +43,14 @@ export function formatTokenId(tokenId = '', size_ = 4) {
     return `${prefix}${head}...${tail}`
 }
 
-export function formatDomainName(domain: string) {
+export function formatDomainName(domain: string, size = 18) {
```
Comment (guanbinrui) [inline packages/web3-shared/evm/src/helpers/formatter.ts:46]: Please update the unit testing.

File: packages/web3-shared/base/src/utils/currency.ts [hunk 2]
```diff
@@ -58,22 +58,22 @@ export function formatCurrency(
     const resolvedBoundaries = defaults({}, boundaries, DEFAULT_BOUNDARIES)
     const resolvedSymbols = defaults({}, symbols, DEFAULT_CRYPTO_CURRENCY_SYMBOLS)
 
-    const symbol = DEFAULT_CRYPTO_CURRENCY_SYMBOLS[currency]
+    const symbol = currency ? DEFAULT_CRYPTO_CURRENCY_SYMBOLS[currency] : ''
```
Comment (guanbinrui) [inline packages/web3-shared/base/src/utils/currency.ts:61]: Please update the unit testing.

File: packages/web3-providers/src/types/Trending.ts [hunk 3]
```diff
@@ -116,6 +129,39 @@ export namespace TrendingAPI {
         volume_all?: number
     }
 
+    export interface Collection {
```
Comment (guanbinrui) [inline packages/web3-providers/src/types/Trending.ts]: Place it `packages/web3-providers/src/types/NonFungibleToken.ts`.

**Comment by hanszhoou** [comment]
svg fixed

File: packages/web3-shared/evm/src/helpers/formatter.ts (hunk 1 above)
Comment (hanszhoou) [reply packages/web3-shared/evm/src/helpers/formatter.ts:46]: done

File: packages/web3-shared/base/src/utils/currency.ts (hunk 2 above)
Comment (hanszhoou) [reply packages/web3-shared/base/src/utils/currency.ts:61]: done

File: packages/web3-providers/src/types/Trending.ts (hunk 3 above)
Comment (hanszhoou) [reply packages/web3-providers/src/types/Trending.ts]: done

**Pull request closed without merging**